conda/misc.py

```diff
@@ -259,7 +259,7 @@ def clone_env(prefix1, prefix2, verbose=True, quiet=False, index_args=None):
         what = "Package%s " % ('' if len(notfound) == 1 else 's')
         notfound = '\n'.join(' - ' + fn for fn in notfound)
         msg = '%s missing in current %s channels:%s' % (what, context.subdir, notfound)
-        raise CondaRuntimeError(msg)
+        raise ChannelError(msg)
```

Comment:
Use `PackageNotFoundError` here.  And fix up the other uses of that exception so they all have a coherent error message.  Use `location` as an input argument to the exception, where location is either usually a prefix or list of channels it looks like.